samples/matter/common/src/bridge/ble_connectivity_manager.cpp

```diff
@@ -444,30 +448,103 @@ CHIP_ERROR BLEConnectivityManager::Scan(ScanDoneCallback callback, void *context
 	return ret;
 }
 
+void BLEConnectivityManager::Recovery::RemoveRecovered(BLEBridgedDeviceProvider *provider)
+{
+	if (!provider) {
+		return;
+	}
+
+	sys_snode_t *node = sys_slist_peek_head(&Instance().mRecovery.mListToRecover);
+	ListItem *item;
+
+	/* Iterate through the list of providers and remove the requested one. */
+	do {
+		item = CONTAINER_OF(node, __typeof__(*item), mNode);
+
+		if (!item) {
+			return;
+		}
+
+		if (item->mProvider == provider) {
+			sys_slist_find_and_remove(&Instance().mRecovery.mListToRecover, node);
+			Platform::Delete(item);
+			return;
+		}
+		node = sys_slist_peek_next(node);
+	} while (node != nullptr);
+}
+
+bool BLEConnectivityManager::Recovery::IsProviderToRecover(BLEBridgedDeviceProvider *provider)
+{
+	if (!provider || !IsNeeded()) {
+		return false;
+	}
+
+	sys_snode_t *node = sys_slist_peek_head(&Instance().mRecovery.mListToRecover);
+	ListItem *item;
+
+	/* Iterate through the list of providers and check if the given one is on it. */
+	do {
+		item = CONTAINER_OF(node, __typeof__(*item), mNode);
+
+		if (!item) {
+			return false;
+		}
+
+		if (item->mProvider == provider) {
+			LOG_ERR("Provider found on a list to recover");
+			return true;
+		}
+		node = sys_slist_peek_next(node);
+	} while (node != nullptr);
```

Comment:
Can we use Single-linked API for searching: https://docs.zephyrproject.org/apidoc/latest/group__single-linked-list__apis.html#gad6f1014e26d6cf9925d00b4f53076116 ?

Reply:
I haven't noticed this API. It looks good, I will give it a try!